core/drivers/crypto/crypto_api/acipher/rsa.c

```diff
@@ -302,8 +302,10 @@ TEE_Result crypto_acipher_rsaes_encrypt(uint32_t algo,
 			if (ret != TEE_SUCCESS)
 				return ret;
 
-			if (msg_len >
-			    rsa_data.key.n_size - 2 * rsa_data.digest_size - 2)
+			if ((msg_len > rsa_data.key.n_size -
```

Comment:
This would be more readable if you kept the original if and only added a new one for
``` C
2 * rsa_data.digest_size >= rsa_data.key.n_size - 2
```
By the way I'd prefer if this test is done before the original test.